### ISSUE TITLE
SOLVED: SSD Enclosures with Jmicron chips JMS580 and JMS583 don't boot up. RPi4.

### ISSUE THREAD

**Issue by carras** [raspberrypi/rpi-eeprom#266]
Hi there:

With the latest EEPROM firmware. I've try critical, stable and beta channels. My SSD enclosure from ORICO using the chip **JMS580** from Jmicron doesn't boot up. And I've seen other people with the same issue with enclosure with **JMS580**.

If we plug the raspi on and after we connect the ssd, then it boots up.

And It was working before, but the latest critical and stable don't work now.

PD: I want to boot up **HassOS** form the SSD, but the implemented an auto firmware (EEPROM) update that it's always going to prevent me to use an older firmware.

Update: I add JMS578 too, to de list of controllers with the same issue.

TRY THIS SOLUTION:

**Solution**: with the command "sudo -E rpi-eeprom-config --edit" you edit the EEPROM BOOTLOADER then you the next configuration:
USB_MSD_PWR_OFF_TIME=0

**Comment by Holland1** [comment]
Can confirm there is a similar issue with the JMS583 controller.

**Comment by caro7372** [comment]
I have exactly the same problem as above described.

**Comment by timg236** [comment]
Please can you add screenshots showing the failure on the bootloader HDMI diagnostics screen or UART traces.

**Comment by lurch** [comment]
>  I want to boot up HassOS form the SSD, but the implemented an auto firmware (EEPROM) update that it's always going to prevent me to use an older firmware.

You could try setting the `FREEZE_VERSION` option as a workaround to stop that? https://www.raspberrypi.org/documentation/hardware/raspberrypi/bcm2711_bootloader_config.md

**Comment by timg236** [comment]
Yes FREEZE_VERSION will prevent upgrades except for rescue image and because it's in the EEPROM will work if you swap SD cards.

The most likely explanation is that the controller is fussy about exactly how long USB port power is off for and any delays during initialisation. However, we'd need screenshots / logs to confirm. 

N.B. Please can  you also confirm that when booted from SD card the enclosure works reliably when mounted under Linux and will hotplug correctly if you connect/disconnect the cable without pressing any special power buttons on the enclosure?

**Comment by carras** [comment]
@timg236 This is it, but it's not Raspberry Pi OS. I can do it with it if you ask, this is HassOS. Is this ok?

![IMG_0329](https://user-images.githubusercontent.com/5551391/102211884-93e24300-3ed4-11eb-8def-4b377aba16d4.jpeg)

The disk enclosure Is visible when you boot Raspberry Pi OS and remember "If we plug the raspi on and after we connect the ssd, then it boots up."

**Comment by carras** [comment]
I add JMS578 too, to de list of controllers with the same issue. As being reported by other user.

**Comment by carras** [comment]
I just contacted Jmicron in case the need something to add, or they can help with the next message:

When try to boot up from USB enclosure on a Raspberry Pi 4 your controllers JMS578, JMS580 and JMS583 are reported to be not working properly.

I've open an issue in the Github page of the EEPROM firmware of the Raspberry Pi 4:
https://github.com/raspberrypi/rpi-eeprom/issues/266

There are many Raspberry Pi 4 users in world that would like to boot up with USB on a SSD and your chips are in many cases like ORICO (which I have two) or Kingspecs. As te result of your controller no working those brands and your brand is losing popularity over Realtech or ASM with apparently work better.

Could you please investigate the issue and help the Raspberry Pi team to work on this problem?

Thanks.

**Comment by timg236** [comment]
I doubt this is anything to do with the JMicron chip because there are plenty of people on the forums using JMicron USB SATA adapters. Some of the older chips have issues with UAS but that's not relevant until Linux runs.
I think the issue is with the design of the enclosure.

**Comment by caro7372** [comment]
> Please can you add screenshots showing the failure on the bootloader HDMI diagnostics screen or UART traces.

![Uploading 8ADFD579-C95C-49DB-AAEA-238F98990404.jpeg…]()

**Comment by caro7372** [comment]
> Yes FREEZE_VERSION will prevent upgrades except for rescue image and because it's in the EEPROM will work if you swap SD cards.
> 
> The most likely explanation is that the controller is fussy about exactly how long USB port power is off for and any delays during initialisation. However, we'd need screenshots / logs to confirm.
> 
> N.B. Please can you also confirm that when booted from SD card the enclosure works reliably when mounted under Linux and will hotplug correctly if you connect/disconnect the cable without pressing any special power buttons on the enclosure?

Yes, SD card works! And cable SSD out and in again boot works.

**Comment by timg236** [comment]
Please post links to the SSD enclosure. If it's still for sale and we should be able to get one for some interop testing in the new year. Please also confirm the exact EEPROM version that worked for you. It may be that a configurable timeout quirk could be added to support such devices.

**Comment by Holland1** [comment]
The one with the JMS583 chip

[https://www.amazon.nl/dp/B07QN2WFGB/ref=pe_19967891_404437601_TE_item](url)

**Comment by caro7372** [comment]
https://www.aliexpress.com/item/4001278381663.html

**Comment by vodoc** [comment]
Small investigation of my SSD (based on JMS583)
https://aliexpress.ru/item/4000402460407.html
EPPROM versions:
Thu  3 Sep 12:11:43 UTC 2020 (1599135103) - boots ok.
Wed 28 Oct 17:32:40 UTC 2020 (1603906360) - boots only after unplug + plug again
Fri 11 Dec 11:15:17 UTC 2020 (1607685317) - boots only after unplug + plug again

**Comment by carras** [comment]
I've got this bundle which I think its a really good quality/price:
https://www.aliexpress.com/item/1005001413198976.html?spm=a2g0s.9042311.0.0.27424c4dTdwJ3z

It use JMS580 and a SATA m.2 SSD.

**Comment by carras** [comment]
@caro7372 I have this Orico enclosure with the JMS578 but it has the format of 2.5 inches disks. And it works perfectly to boot up with the Raspberry. Can you check with other ssd blade?
https://www.aliexpress.com/item/4001146259311.html?spm=a2g0s.9042311.0.0.27424c4dc8LdKF
*In that enclosure I put a Samsung 540 SSD.

**Comment by timg236** [comment]
Has anyone tried increasing the USB port power-off time by setting USB_MSD_PWR_OFF_TIME=5000 in the EEPROM config? This should look identical to a disconnect to the device. The default off-time may be less between 3rd September and subsequent releases.

**Comment by caro7372** [comment]
I will check and try!

**Comment by carras** [comment]
> Has anyone tried increasing the USB port power-off time by setting USB_MSD_PWR_OFF_TIME=5000 in the EEPROM config? This should look identical to a disconnect to the device. **The default off-time may be less between 3rd September and subsequent releases.**

Well, this made the trick. **I didn't make it 5000 I made it 0**. Then it works. I've tried your 5000 suggestion but It didn't work.

**Solution**: with the command "sudo -E rpi-eeprom-config --edit" you edit the EEPROM BOOTLOADER then you the next configuration:
USB_MSD_PWR_OFF_TIME=0

@timg236 I guess there is a reason for it. How much USB_MSD_PWR_OFF_TIME=1000 is necesary? Is it bad to have it at 0? Why is important to cut the power to the USB for some time before it boots?

From release notes:
"USB ports are now powered off for 1 second at startup regardless of boot mode to resolve issues with some USB devices which failed after a reboot. This can be configured via the existing USB_MSD_PWR_OFF_TIME EEPROM setting."
https://github.com/raspberrypi/rpi-eeprom/releases/tag/v2020.07.31-138a1

The documentation can be found:
https://www.raspberrypi.org/documentation/hardware/raspberrypi/booteeprom.md
https://www.raspberrypi.org/documentation/hardware/raspberrypi/bcm2711_bootloader_config.md

**Comment by caro7372** [comment]
Where did you set that 0 ms?
In the config.txt ( on the SD-card ) or ???? I'm a bit confused now if I did the right thing.

**Comment by carras** [comment]
> @carras
> Where did you set that 0 ms?
> In the config.txt ( on the SD-card ) or ???? I'm a bit confused now if I did the right thing.

You get an SD-card with Raspberry Pi OS and boot with it.
Then you execute the command in SSH: sudo -E rpi-eeprom-config --edit
You are now in one text file and you need to add this at the end of the list: USB_MSD_PWR_OFF_TIME=0

**Comment by caro7372** [comment]
> > @carras
> > Where did you set that 0 ms?
> > In the config.txt ( on the SD-card ) or ???? I'm a bit confused now if I did the right thing.
> 
> You get an SD-card with Raspberry Pi OS and boot with it.
> Then you execute the command in SSH/terminal: sudo -E rpi-eeprom-config --edit
> You are now in one text file and you need to add this at the end of the list: USB_MSD_PWR_OFF_TIME=0

What I see know is that the 2020-10-28 has no stable release?
2020-12-11 is latest stable.
You see that too?

After applying the USB ...... to 0 you reboot with or without the SD card in ( for the first time ? ).
Sorry for my noob questions but I'm relatively new to Linux.

**Comment by timg236** [comment]
USB port power is not guaranteed across a reboot, it goes away briefly on 2G, 4G during reset and for much longer on 8G. This is the same as a disconnect and for devices such as spinning hard drive enclosures, it's better to have a long power off than a short one. For an NVMe SSD it should be fine to have USB_MSD_PWR_OFF_TIME=0 because it will power on almost immediately.

So changes to improve compatibility for spinning hard drive enclosures seems to be worse for some JMicron NVME adapters. There might be a common change which is good for both but NVME enclosures have been fairly low down the compatibility list because there's no performance benefit of USB SATA SSD

**Comment by timg236** [comment]
> > > @carras
> > > Where did you set that 0 ms?
> > > In the config.txt ( on the SD-card ) or ???? I'm a bit confused now if I did the right thing.
> > 
> > 
> > You get an SD-card with Raspberry Pi OS and boot with it.
> > Then you execute the command in SSH/terminal: sudo -E rpi-eeprom-config --edit
> > You are now in one text file and you need to add this at the end of the list: USB_MSD_PWR_OFF_TIME=0
> 
> What I see know is that the 2020-10-28 has no stable release?
> 2020-12-11 is latest stable.
> You see that too?
> 
> After applying the USB ...... to 0 you reboot with or without the SD card in ( for the first time ? ).
> Sorry for my noob questions but I'm relatively new to Linux.

2020-10-28 is the version number of a beta release. Stable releases are a subset of beta releases and default releases (and critical updates) are a subset of stable releases. Therefore, if you configure your system to track the beta releases you'll see all the images, some good, some bad.

**Comment by caro7372** [comment]
> > > > @carras
> > > > Where did you set that 0 ms?
> > > > In the config.txt ( on the SD-card ) or ???? I'm a bit confused now if I did the right thing.
> > > 
> > > 
> > > You get an SD-card with Raspberry Pi OS and boot with it.
> > > Then you execute the command in SSH/terminal: sudo -E rpi-eeprom-config --edit
> > > You are now in one text file and you need to add this at the end of the list: USB_MSD_PWR_OFF_TIME=0
> > 
> > 
> > What I see know is that the 2020-10-28 has no stable release?
> > 2020-12-11 is latest stable.
> > You see that too?
> > After applying the USB ...... to 0 you reboot with or without the SD card in ( for the first time ? ).
> > Sorry for my noob questions but I'm relatively new to Linux.
> 
> 2020-10-28 is the version number of a beta release. Stable releases are a subset of beta releases and default releases (and critical updates) are a subset of stable releases. Therefore, if you configure your system to track the beta releases you'll see all the images, some good, some bad.

Yes, I understand that ( now ) but was a bit confused about Home Assistant 5.8 was setting it to 2020-10-28 although it now seems it was Beta.

**Comment by carras** [comment]
@caro7372 There are 3 types of releases that you can subscribe to: "critical" "stable" and "beta".
I recommend you use the command "sudo raspi-config" go to advance options, then bootloader version, and the select either latest or stable. Then you need to edit the file I told you about.

Or just do the file thing, because HassOS will update the firmware for you apparently.

**Comment by timg236** [comment]
It's up to the Home Assistant maintainers to decide which bootloader updates the OS installs. Tracking beta by default seems rather risky because the whole point of beta is to find bugs!

**Comment by carras** [comment]
> There might be a common change which is good for both but NVME enclosures have been fairly low down the compatibility list because there's no performance benefit of USB SATA SSD

@timg236 Yes, but I use a SATA m.2 enclosure, no NVMe. But as I said before I feel this issue is only on Jmicron controllers. So do I have to update the config file with every update?
*I've tested that it doesn't work with "critical" updates, only from stable.

**Comment by caro7372** [comment]
> @caro7372 There are 3 types of releases that you can subscribe to: "critical" "stable" and "beta".
> I recommend you use the command "sudo raspi-config" go to advance options, then bootloader version, and the select **latest**. Then you need to edit the file I told you about.
> 
> Or just do the file thing, because HassOS will update the firmware for you apparently.

Yes, that part I'm ( was ) aware of. So I set it back to "stable".
I'm setting up the system again!

**Comment by vodoc** [comment]
I may confirm that USB_MSD_PWR_OFF_TIME=0 works fine for my SSD (based on JMS583) on latest beta Fri 11 Dec 11:15:17 UTC 2020 (1607685317)
It if not fixes boot issue for HassOS (it replaces config each boot), but looks like we may have solution and fix in the future.

**Comment by timg236** [comment]
@vodoc Thanks for the tests. I've requested this and some other devices for the interop test but that won't happen until the New Year. Hopefully, we'll find something which works with both these drives and the library of awkward devices :)

N.B. If HassOS is using rpi-eeprom-update then it should respect FREEZE_VERSION=1 in the bootloader EEPROM config and leave the bootloader alone.

**Comment by carras** [comment]
@vodoc Wait a moment, HassOS it does work with me. On reboot, soft reboot and turn off and on. Check the issue there: https://github.com/home-assistant/operating-system/issues/1045

@timg236 Ok, let's assume FREEZE_VERSION=1 works. But do I lost the config file every time it updates? Can I prevent the config file to be modify?

And, another question, I put FREEZE_VERSION=1 but I deleted later, the documentation said something about how to disable different that remove it from config file. Do I have to do another thing to disable it and that's why @vodoc can't reboot without resetting the config file?

Documentation: FREEZE_VERSION

Previously this property was only checked by the rpi-eeprom-update script. However, now that self-update is enabled the bootloader will also check this property. If set to 1, this overrides ENABLE_SELF_UPDATE to stop automatic updates. **To disable** FREEZE_VERSION you will have to use an SD card boot with recovery.bin.

**Comment by agners** [comment]
@caro7372  @timg236 yeah we were debating if we should use beta, but we thought since it fixes known issues of external hard drives (and the update only applies if it is on a USB hard drive), it make sense to take the plunge (see discussion here https://github.com/home-assistant/operating-system/pull/939).

Maybe it was a bad decision... But then, the latest stable would now have broken JMS580/JMS583 too :man_shrugging: 

In the end, I am inclined to just remove firmware update entirely, then users can figure out what firmware/USB SSD combinations work using Raspberry Pi OS, and then switch to Home Assistant OS once they have a working firmware/USB SSD combination.

**Comment by caro7372** [comment]
Goodmorning, I read the discussion about auto update the eeprom. Only thing I am curious about is why HomeAssistant wants to update this.
I’m sure there will be benefit of doing it but is it really needed?

Regards,

Robert

**Comment by p4p3r** [comment]
In case it's useful for future testing, here's a few enclosures I tested (RPi4, HassOS 64bit):
Booting fine:
- Inateck FEU3NS-1E, reported as ASM1153E VendorID: 0x174c Product ID: 0x55aa
- Inateck FE2005, advertised as using JMicron JMS567, reported as JMicron Vendor ID: 0x152d Product ID: 0x0578
- POSUGEAR HD-SATA, reported as JMicron JMS579 VendorID 0x152d ProductID 0x0578
Booting only after unplug/plug:
- Inateck FE2010, advertised as using JMicron JMS578, reported as Toshiba VendorID: 0x0080 ProductID: 0xa001

**Comment by THX1180** [comment]
Having a similar with my Orico bh100 external SSD (JMicron JMS567 SATA 6Gb/s) and opened a thread on the official forums...

Here's what a typed out on the forum:

Hi Raspberry Pi community,

I bought a powered hub ( TP-LINK UH720) as recommended in the storage quirks thread) to boot with my external SSD (Orico bh100), because i was heaving power issues connecting the SSD directly to the Pi. I reinstalled everything from scratch and everything booted up fine, and in fact worked like a charm for weeks on end.

No error messages or strange behavior whatsoever...

Updated the system a couple of days ago and it's no longer functioning properly (see the included images). I know the SSD is working fine because, i have extensively tested it on the PC (manjaro) and it was functioning properly for weeks. The Pi i directly connected and powered by the hub and it's getting enough juice, because i can easily connected other external hdd's etc... But SSD will often just not even register, the hub has little lights (indicators of connection) and it will just blink quickly and then do nothing. It will sporadically boot up into the system when i power it on/off randomly 6 to10 times but that's about it, when i try to reboot i get the error messages and it won't do jacks$#5

It will just hang displaying these messages (click to enlarge the images):

![20201216-145348](https://user-images.githubusercontent.com/56322778/102493982-44d31400-4074-11eb-9560-ba6f08678d26.jpg)
![20201216-145353](https://user-images.githubusercontent.com/56322778/102494135-7fd54780-4074-11eb-919d-6a666cef105e.jpg)

I already did a fresh install and add the quirks, and even tried manjaro instead of Pi Os. But nothing and i mean nothing seems to work, and if it works it will stop working when rebooting.

SD card booting works without any issues...

I noticed someone posted a vid on Youtube with the same issue, he has a normal harddrive connected to his Pi but with a separate power switch for his drive:

https://www.youtube.com/watch?v=XLyQY1J85-c

What can i do to resolve this problem??? Please help me out...

My system specs:

Rpi 4 4Gb
Rpi OS 32-bit
TP-LINK UH720 powered USB hub
Orico-bh100 external SDD

Can't share logs because... Can't boot into system. :shock:

Thanks in advance!!! :(

Mind you this is an external SSD not an external SSD adapter... 

Here's a link to the thread i opened on the official forums:

https://www.raspberrypi.org/forums/viewtopic.php?f=28&t=295119

**Comment by carras** [comment]
@THX1180 The solution is in the first comment and in this comment: https://github.com/raspberrypi/rpi-eeprom/issues/266#issuecomment-746890101

**Comment by THX1180** [comment]
> @THX1180 The solution is in the first comment and in this comment: [#266 (comment)](https://github.com/raspberrypi/rpi-eeprom/issues/266#issuecomment-746890101)

Yeah it's not working for me... I have to plug/unplug multiple times, the SSD boots when it wants to. 

![20201217_173433](https://user-images.githubusercontent.com/56322778/102516684-6ee5ff80-408f-11eb-8c10-5fbdb65fe073.jpg)
![20201217_173433](https://user-images.githubusercontent.com/56322778/102516698-73121d00-408f-11eb-9cbc-2243e7923652.jpg)

What is causing this???

**Comment by carras** [comment]
@THX1180 do the solution and try with the disk attach with no hub.

**Comment by lurch** [comment]
@carras Just because your solution works for you, I don't think it's safe to assume that it'll always work for everyone else? :shrug: 
(If it was really "one size fits all", I assume @timg236 would have already been doing that in the EEPROM code!)

**Comment by THX1180** [comment]
> @THX1180 do the solution and try with the disk attach with no hub.

Yeah that doesn't work... Same problem.

**Comment by carras** [comment]
@lurch give him something better. Just trying to help with what I know. 

@THX1180 your case has the same controller, if I recommend to star over. Make sure your confi file keep with the custom config after reboots. Can you copy here your config file after a reboot?

**Comment by THX1180** [comment]
> @lurch give him something better. Just trying to help with what I know.
> 
> @THX1180 your case has the same controller, if I recommend to star over. Make sure your confi file keep with the custom config after reboots. Can you copy here your config file after a reboot?

Thanks for helping me... Do you mean the eeprom config file???

I already did a fresh install of everything, and applied the eeprom config fix but no change.

**Comment by carras** [comment]
@THX1180 Yes, when you do the command of the solution there is a kind of text file. After a reboot copy here the text in there.

**Comment by THX1180** [comment]
> @THX1180 Yes, when you do the command of the solution there is a kind of text file. After a reboot copy here the text in there.

```
[all]
BOOT_UART=0
WAKE_ON_GPIO=1
POWER_OFF_ON_HALT=0
USB_MSD_PWR_OFF_TIME=0
```

**Comment by carras** [comment]
@THX1180 I think you don't have the latest firmware. Use te command `vcgencmd bootloader_version` and you want this version date: Dec 11 2020

And my config is:
```
[all]
BOOT_UART=0
WAKE_ON_GPIO=1
POWER_OFF_ON_HALT=0
HDMI_DELAY=0
USB_MSD_PWR_OFF_TIME=0
```

Also check this comment, cuz it's maybe a different approach: https://github.com/home-assistant/operating-system/issues/1045#issuecomment-747592332

But I think if you get the latest EEPROM and `USB_MSD_PWR_OFF_TIME=0` you will solve the problem. But idk with your JMS

You can also try changing values of `USB_MSD_PWR_OFF_TIME=0`. I've tried 5000, 1500, 1000, 500 and 0. And only 0 works for me. Make sure you reboot before any change and then "sudo shutdown now" (idk if there is any different, but I do that) then turn off power connect the SSD enclosure, remove SD card and connect the power.

**Comment by THX1180** [comment]
> @THX1180 I think you don't have the latest firmware. Use te command `vcgencmd bootloader_version` and you want this version date: Dec 11 2020
> 
> And my config is:
> 
> ```
> [all]
> BOOT_UART=0
> WAKE_ON_GPIO=1
> POWER_OFF_ON_HALT=0
> HDMI_DELAY=0
> USB_MSD_PWR_OFF_TIME=0
> ```
> 
> Also check this comment, cuz it's maybe a different approach: [home-assistant/operating-system#1045 (comment)](https://github.com/home-assistant/operating-system/issues/1045#issuecomment-747592332)
> 
> But I think if you get the latest EEPROM and `USB_MSD_PWR_OFF_TIME=0` you will solve the problem. But idk with your JMS
> 
> You can also try changing values of `USB_MSD_PWR_OFF_TIME=0`. I've tried 5000, 1500, 1000, 500 and 0. And only 0 works for me. Make sure you reboot before any change and then "sudo shutdown now" (idk if there is any different, but I do that) then turn off power connect the SSD enclosure, remove SD card and connect the power.

Thank you carras... I will your suggestions and report back!!!

Thank you for helping me out...👍👍👍

**Comment by timg236** [comment]
Managed to get an Orico (JMicron) NVMe SSD  and an Elutang (RealTek). As expected the RealTek works and the JMicron seems to be fussy about the software-controlled USB port power off which is why USB_MSD_PWR_OFF_TIME=0 works.

For the moment the September 3rd bootloader (default in RPI Imager Rescue) seems to work or adding USB_MSD_PWR_OFF_TIME=0. Once the JMicron behaviour is fully understood then hopefully the generic change can be made but that won't be for a few weeks.

N.B. This shouldn't be an issue on the 8GB which has different power sequencing at boot.

**Comment by THX1180** [comment]
> Managed to get an Orico (JMicron) NVMe SSD and an Elutang (RealTek). As expected the RealTek works and the JMicron seems to be fussy about the software-controlled USB port power off which is why USB_MSD_PWR_OFF_TIME=0 works.
> 
> For the moment the September 3rd bootloader (default in RPI Imager Rescue) seems to work or adding USB_MSD_PWR_OFF_TIME=0. Once the JMicron behaviour is fully understood then hopefully the generic change can be made but that won't be for a few weeks.
> 
> N.B. This shouldn't be an issue on the 8GB which has different power sequencing at boot.

Really awesome to hear this news... So it has to do with power sequencing? How come it works instantaneously when pulling the plug, and leaving it out for an x amount seconds when replugging? (Mind you I am talking about when using a powered hub)...

Same thing when I use the button on the hub, and don't power on/off/on too fast. It's like something is kept in memory, or power needs to drain completely after it's turned off...

Could you test with the Jmicron and a powered hub? Boot/reboot turn on and off x amount of times.. 

Cheers!

**Comment by timg236** [comment]
I've attached an experimental change which works for me with the Orico and the other awkward devices which I have at home. It won't get released until I can get back in the lab in the new year and ideally look at some of these devices with a scope. 

However, feel free to post interop reports here which will give us more (or less) confidence about releasing this.

[rpi-eeprom-recovery-nvme.zip](https://github.com/raspberrypi/rpi-eeprom/files/5719205/rpi-eeprom-recovery-nvme.zip)

BOOTLOADER_VERSION: Dec 19 2020 4e126d29

Flash the bootloader using the Raspberry Pi Imager "custom image" with the attached zip

The issue seems to be that some USB devices get into a bad state during the chip-reset on Pi 4 1,24 GB . Unplugging them normally resolves this so the bootloader uses the port power control to simulate this.  The problem is that to do this it actually has to initialise the XHCI controller which causes some activity. So the generic solution is finding the best sequencing of XHCI reset + power on.

**Comment by THX1180** [comment]
> I've attached an experimental change which works for me with the Orico and the other awkward devices which I have at home. It won't get released until I can get back in the lab in the new year and ideally look at some of these devices with a scope.
> 
> However, feel free to post interop reports here which will give us more (or less) confidence about releasing this.
> 
> [rpi-eeprom-recovery-nvme.zip](https://github.com/raspberrypi/rpi-eeprom/files/5719205/rpi-eeprom-recovery-nvme.zip)
> 
> BOOTLOADER_VERSION: Dec 19 2020 4e126d29
> 
> Flash the bootloader using the Raspberry Pi Imager "custom image" with the attached zip
> 
> The issue seems to be that some USB devices get into a bad state during the chip-reset on Pi 4 1,24 GB . Unplugging them normally resolves this so the bootloader uses the port power control to simulate this. The problem is that to do this it actually has to initialise the XHCI controller which causes some activity. So the generic solution is finding the best sequencing of XHCI reset + power on.

Awesome! 

I will try this out and i will report back asap!!!!

**Comment by vodoc** [comment]
```
pi@raspberrypi:~ $ sudo rpi-eeprom-update
BCM2711 detected
Dedicated VL805 EEPROM detected
BOOTLOADER: up-to-date
CURRENT: Sat 19 Dec 10:06:09 UTC 2020 (1608372369)
 LATEST: Fri 11 Dec 11:15:17 UTC 2020 (1607685317)
 FW DIR: /lib/firmware/raspberrypi/bootloader/beta
VL805: up-to-date
CURRENT: 000138a1
 LATEST: 000138a1
pi@raspberrypi:~ $ sudo rpi-eeprom-config
[all]
BOOT_UART=0
WAKE_ON_GPIO=1
POWER_OFF_ON_HALT=0
```
SSD based on JMS583. boots fine.
Thank you very much!

**Comment by timg236** [comment]
> > I've attached an experimental change which works for me with the Orico and the other awkward devices which I have at home. It won't get released until I can get back in the lab in the new year and ideally look at some of these devices with a scope.
> > However, feel free to post interop reports here which will give us more (or less) confidence about releasing this.
> > [rpi-eeprom-recovery-nvme.zip](https://github.com/raspberrypi/rpi-eeprom/files/5719205/rpi-eeprom-recovery-nvme.zip)
> > BOOTLOADER_VERSION: Dec 19 2020 4e126d29
> > Flash the bootloader using the Raspberry Pi Imager "custom image" with the attached zip
> > The issue seems to be that some USB devices get into a bad state during the chip-reset on Pi 4 1,24 GB . Unplugging them normally resolves this so the bootloader uses the port power control to simulate this. The problem is that to do this it actually has to initialise the XHCI controller which causes some activity. So the generic solution is finding the best sequencing of XHCI reset + power on.
> 
> Awesome!
> 
> I will try this out and i will report back asap!!!!
> 
> Update:
> 
> The problems occurs less frequently now, so that's really an improvement. it's not resolved but we are getting there...👍
> 
> Thanks for all of your hard work!!!!

Interesting. It’s survived a few hundred reboots for me but there might be some variation between boards and cables. Looks like I’ll need to use the scope after all!

**Comment by sanyafifa** [comment]
Guys, hello!
After the update, it stopped booting from the SSD disk.
It starts if you turn off / turn on the SSD while the raspberry is running, the red and green diodes are constantly on.

`vcgencmd bootloader_version
Dec 11 2020 11:15:17

lsusb
Bus 002 Device 002: ID 152d:1576 JMicron Technology Corp. / JMicron USA Technology Corp.`

![image](https://user-images.githubusercontent.com/49896945/103137475-771aec00-46da-11eb-8730-2f4bd59e048a.png)

`sudo -E rpi-eeprom-config --edit

[all]
BOOT_UART=0
WAKE_ON_GPIO=1
POWER_OFF_ON_HALT=0
DHCP_TIMEOUT=45000
DHCP_REQ_TIMEOUT=4000
TFTP_FILE_TIMEOUT=30000
ENABLE_SELF_UPDATE=1
DISABLE_HDMI=0
BOOT_ORDER=0xf41
USB_MSD_PWR_OFF_TIME=0`

**Comment by pvnnpavan** [comment]
I had same issue with orico and jmicron controller and have to revert back to 3 Sep boot loader.

Is "USB_MSD_PWR_OFF_TIME=0" permanent fix or is this workaround ?

Also I noticed difference with boot loaders which got copied into below folder, why are all entries removed in Dec boot loader and how is this going to affect the booting. 

/lib/firmware/raspberrypi/bootloader/stable

-rw-r--r-- 1 root root  98904 Feb 28  2020 vl805-000137ad.bin
-rw-r--r-- 1 root root 524288 Apr 23  2020 pieeprom-2020-04-16.bin
-rw-r--r-- 1 root root 524288 Jun 17  2020 pieeprom-2020-06-15.bin
-rw-r--r-- 1 root root  99224 Jul 20 08:17 vl805-000138a1.bin
-rw-r--r-- 1 root root 524288 Jul 20 08:17 pieeprom-2020-07-16.bin
-rw-r--r-- 1 root root 524288 Aug 10 10:17 pieeprom-2020-07-31.bin
-rw-r--r-- 1 root root 524288 Sep  7 10:35 pieeprom-2020-09-03.bin
-rw-r--r-- 1 root root 106392 Dec 15 04:51 recovery.bin
-rw-r--r-- 1 root root 524288 Dec 15 04:51 pieeprom-2020-12-11.bin


root@pavanraspi:/lib/firmware/raspberrypi/bootloader/stable# rpi-eeprom-config pieeprom-2020-07-31.bin
[all]
BOOT_UART=0
WAKE_ON_GPIO=1
POWER_OFF_ON_HALT=0
DHCP_TIMEOUT=45000
DHCP_REQ_TIMEOUT=4000
TFTP_FILE_TIMEOUT=30000
ENABLE_SELF_UPDATE=1
DISABLE_HDMI=0
BOOT_ORDER=0xf41

root@pavanraspi:/lib/firmware/raspberrypi/bootloader/stable# rpi-eeprom-config pieeprom-2020-09-03.bin
[all]
BOOT_UART=0
WAKE_ON_GPIO=1
POWER_OFF_ON_HALT=0
DHCP_TIMEOUT=45000
DHCP_REQ_TIMEOUT=4000
TFTP_FILE_TIMEOUT=30000
ENABLE_SELF_UPDATE=1
DISABLE_HDMI=0
BOOT_ORDER=0xf41

root@pavanraspi:/lib/firmware/raspberrypi/bootloader/stable# rpi-eeprom-config pieeprom-2020-12-11.bin
[all]
BOOT_UART=0
WAKE_ON_GPIO=1
POWER_OFF_ON_HALT=0

**Comment by 3avidity** [comment]
I can confirm that my SATA/USB boot stopped working with an update I did yesterday.  Enclosure: ORICO TCM2F-C3 NGFF to USB3.1 found at https://www.newegg.com/orico-tcm2f-c3/p/0VN-0003-001G9, using JMS580 controller.

I can also confirm that once I set POWER_OFF_ON_HALT=0, then reboots started working again.  

Symptoms: When power is applied to the RPi4 the blue SATA enclosure light comes on bright (normal) and then a split second later goes into a very dim (not normal) output.  This indicates to me that the enclosure thinks it's in a "power standby" mode.  In this state the Pi cannot use it to boot off of because it's not responding, when the Pi thinks it should.

I have a question: Will future updates to the firmware settings re-set this back to 1000ms?

**Comment by llooping** [comment]
I confirm the issue also with JMS577/9 Series (But use JMS578 firmware)

I will do the test with USB_MSD_PWR_OFF_TIME=0 to see if ti works well or not

**Comment by timg236** [comment]
@vodoc @THX1180 

There's an updated version of the generic USB power off changes here. I left it running in a loop and survived a few thousand reboots.
[rpi-eeprom-recovery-nvme-v3.zip](https://github.com/raspberrypi/rpi-eeprom/files/5763969/rpi-eeprom-recovery-nvme-v3.zip)

N.B For the current stable/beta releases using a USB3 HUB seems to make the JMicron chip happier. The Realtek and AsMedia devices worked fine with both releases.

@3avidity The rpi-eeprom-update automatic update process preserves the bootloader configuration so you would only need to re-apply the changes if you used the Raspberry Pi Imager to revert to factory settings.

**Comment by THX1180** [comment]
> @vodoc @THX1180
> 
> There's an updated version of the generic USB power off changes here. I left it running in a loop and survived a few thousand reboots.
> [rpi-eeprom-recovery-nvme-v3.zip](https://github.com/raspberrypi/rpi-eeprom/files/5763969/rpi-eeprom-recovery-nvme-v3.zip)
> 
> N.B For the current stable/beta releases using a USB3 HUB seems to make the JMicron chip happier. The Realtek and AsMedia devices worked fine with both releases.
> 
> @3avidity The rpi-eeprom-update automatic update process preserves the bootloader configuration so you would only need to re-apply the changes if you used the Raspberry Pi Imager to revert to factory settings.

First of all Happy New Year!!!

Thanks for uploading this! Going to test it out and report back with my experience.

Cheers!

**Comment by timg236** [comment]
@THX1180 Does USB_MSD_PWR_OFF_TIME=0 resolve this for you? For some people it does, but not others.

**Comment by timg236** [comment]
As a heads-up the attached version will be the next official beta release. After some more experiments with known awkward hardware (Geekworm, Microsoft Wireless keyboard, Orico NVME) I went back to something similar to the September release with the minimal changes for slow to init devices (Geekworm)

This issue has become a little confused because there are minor differences between board revisions hubs etc so people are seeing different behaviour. Therefore, once it's released I'll close this and ask people to fork specific bugs, personally I'd rather have one bug per hardware setup and close as duplicate once we are sure than debug two things at once :)

[rpi-eeprom-recovery-nvme-v4.zip](https://github.com/raspberrypi/rpi-eeprom/files/5769964/rpi-eeprom-recovery-nvme-v4.zip)

**Comment by THX1180** [comment]
> @THX1180 Does USB_MSD_PWR_OFF_TIME=0 resolve this for you? For some people it does, but not others.

Nope it was already set but it doesn't change anything. 

Guess I'll have to get by with this for now....

Thanks for all of your hard work!!!

**Comment by 3avidity** [comment]
Interesting ... My Orico NVMe with JMS583 did not have the problem, while the Orico SATA/JMS580 did.  I think the brand really won't matter too much, as the complexity of solving this completely depends on the master chip used, and the version of firmware.  I have a number of SSD enclosures with varying chips.  I'd be happy to spend some time trying them if you think this would help you.

**Comment by timg236** [comment]
@3avidity The brand is important because manufacturers because in addition to the chipset external power supplies, buttons can change behaviour e.g. I've had some enclosures which require the power button to be pressed after a reboot (USB power toggle) making it unsuitable for use as a boot device.

**Comment by 3avidity** [comment]
@timg236 In general, I agree with you.  I was looking at it from the perspective that device(s) that had been working before the December update suddenly stopped working.  So unsuitable devices would not have been being used for this purpose prior to the December update.  In my software development experience a priority has been "Don't break what has been working."  It's embarrassing to explain to customers why something they have come to depend on suddenly is broken.  When it comes to "new" features, it is less painful as users have not had time to come to depend on it, and are more forgiving as they understand that sometimes "things just happen."

I was just suggesting I could bring to table results of devices I possess, which you may not have available to you.

**Comment by timg236** [comment]
@3avidity Interop reports are welcome, I was just asking for enough information that someone could reproduce the entire setup if they wanted. 

The "power button" behaviour is an example of a device which has **never** worked (and never will) because it assumes that USB power will never go away during a reset which is not possible with the Pi 4B hardware. The brand and model was significant because other enclosures using the same chipset behaved correctly.

This 'issue' is addressing support for devices which typically fail by not enumerating or hanging after a reset on R1.0, R1.1 Pi4 B boards (not relevant on 1.4/8GB Pi400). The 5th Dec release is pretty much identical to the Sep release and works with all the known awkward devices including those which failed with Sep release. If you know of an exception please let us know because this is going to be frozen very soon.

After that, if a device enumerates but hangs then generic code to reset endpoints etc might be improved. If it fails to enumerate then it's invisible to software and rather than tweaking generic behaviour, I think the answer will likely involve some EEPROM configurable delay (if it helps) or maybe forced PMIC reset.

**Comment by llooping** [comment]
Hi, I can confirm the Solution: with the command "sudo -E rpi-eeprom-config --edit" you edit the EEPROM BOOTLOADER then you the next configuration:
USB_MSD_PWR_OFF_TIME=0

Works for me with a JMS577/9 with a firmware based on JMS578.

Thank you guys

**Comment by timg236** [comment]
Has anyone managed to try the Dec 5th release without USB_MSD_PWR_OFF_TIME=0? Just trying to get an idea of on whether the factory-default settings now support the maxmium possible set of devices.

N.B. There is an unrelated failure (perhaps 1 in 1000 reboots) which also occurs with this device on Sep release. USB boot starts but a SCSI read hangs. I'll look at that separately to see whether it can be recovered. From what I've seen and what I can tell from the comments this 'issue' fails virtually every time.

**Comment by 3avidity** [comment]
@timg236 The Dec 5th release, I believe, is what broke my Pi 4 bootup.  Applying the '0' setting then allowed it to boot up.

**Comment by llooping** [comment]
Same here, I updated my PI 2 weeks ago and It was broken and Now the the the line + 0 it works again :)

**Comment by timg236** [comment]
I released a new beta yesterday (12 Jan) which works with Oricio NVMe with Pi 4b R1.0, R1.1, R1.4 with/without USB_MSD_PWR_OFF_TIME set to zero.

This will be the next stable release unless a major regression is found because the default configuration improves interop over 11th Dec release.

N.B. The default Sep release does actually power off the ports but it looks as though the VLI is a little
timing-sensitive which is why the behaviour changed in the Dec stable release. Setting this to zero avoids any USB PWR control which is good for some devices but bad for others.

**Comment by sanyafifa** [comment]
> I released a new beta yesterday (12 Jan)

it worked for me, thanks!
`Bus 002 Device 002: ID 152d:1576 JMicron Technology Corp. / JMicron USA Technology Corp.`

**Comment by timg236** [comment]
@sanyafifa Thanks for the test. Please could you update your comment to link to the USB device as well of the chipset? It helps others including RPi reproduce the setup in the future.

**Comment by eboeziek** [comment]
Hello, I have a raspberry 3B+ with the Orico BH100, which as the:
Bus 001 Device 005: ID 152d:0567 JMicron Technology Corp. / JMicron USA Technology Corp. JMS567 SATA 6Gb/s bridge

Until yesterday it worked fine, however after an update it won't boot anymore.
I tried what was mentioned above:
sudo -E rpi-eeprom-config --edit
USB_MSD_PWR_OFF_TIME=0

also in my config.txt I already had and have:
program_USB_boot_mode=1

in cmdline.txt I have also tried changing the root=PARTUUID=85f65e63-02 to root=/dev/sda2.

All to no avail.

To be honest I am not sure how to reflash the eeprom back to 09-30 version, if that would help.

The Orico drive now also has issues being recognized on my Mac; diskutil list shows no drive....(yesterday the mac first didn't find it, and then it does; the raspberry pi does see it clearly)
Using an SD I have no issues getting into the Rasperry Pi. Is it possible that the EEPROM update somehow messed up the Orico firmware? lsusb and lsblk do show that the Orico SSD is connected though.

**Comment by pelwell** [comment]
A 3B+ does not have an EEPROM, so editing the EEPROM config will not do anything useful.

**Comment by EtienneB010** [comment]
> A 3B+ does not have an EEPROM, so editing the EEPROM config will not do anything useful.

I just figured that out ;-). Thanks for confirming that.
I do see rpi-eeprom updates coming by when I do sudo update, a bit confusing ;-)

**Comment by pelwell** [comment]
That's because rpi-update (and the apt update mechanisms) is intended to maintain the image for all models of Pi, not just the one you are using. The only exception to this is if you are running an old image without Pi 4 support it won't add it - just in case the boot partition isn't large enough.

**Comment by bschatzow** [comment]
Not sure if this is a good place to post or not.  My issue is I use Home Assistant on the rpi-4 4 GB.  My system boots off the SSD drive an has no issues with OS 5.2 any updates after this causes my system to freeze in 2 hours to 2 days.  No useful information in the logs.  CPU use rises, Memory, and everything else is normal.  My Controller is a StarTech (USB3S2SAT3CB) and the SSD is a Kingston 120GB A400 SATA 3 2.5" Internal SSD SA400S37.  Many comments in https://github.com/home-assistant/operating-system/issues/1119.  I am willing to help test.  Let me know what addition information is needed and what I should try.
Thanks.

**Comment by tablatronix** [comment]
Not sure if this fixes all my ssd issues, but I just rebooted and it booted incredibly fast, I think I was having boot failures before looping. Promising so far.

**Comment by timg236** [comment]
> Not sure if this is a good place to post or not. My issue is I use Home Assistant on the rpi-4 4 GB. My system boots off the SSD drive an has no issues with OS 5.2 any updates after this causes my system to freeze in 2 hours to 2 days. No useful information in the logs. CPU use rises, Memory, and everything else is normal. My Controller is a StarTech (USB3S2SAT3CB) and the SSD is a Kingston 120GB A400 SATA 3 2.5" Internal SSD SA400S37. Many comments in [home-assistant/operating-system#1119](https://github.com/home-assistant/operating-system/issues/1119). I am willing to help test. Let me know what addition information is needed and what I should try.
> Thanks.

That sounds like a Linux kernel update issue because the Linux kernel doesn't depend upon the bootloader USB stack.

**Comment by THX1180** [comment]
Hi guys a quick update (workaround) for those who own the Orico BH100 external SSD. 

This could potentially work for other models/brands as well. But it kind of defeats the purpose of this thread lol.

Check out the gallery:

http://imgur.com/gallery/plL1F0t

I basically replaced the Jmicron controller with a  ASM1153E bridge, it's super easy to do and works flawlessly. Open up the external case by removing the plastic tab on the front/back, slide out the SSD and detach the Jmicron controller. Plug in the ASM (of your choice) controller and slide back into the case.

Mind you it's not secured in the case (need to figure out how to, but it's not necessary to do so).

Jmicron is just not suitable in my opinion.

**Comment by timg236** [comment]
> Hello, I have a raspberry 3B+ with the Orico BH100, which as the:
> Bus 001 Device 005: ID 152d:0567 JMicron Technology Corp. / JMicron USA Technology Corp. JMS567 SATA 6Gb/s bridge
> 
> Until yesterday it worked fine, however after an update it won't boot anymore.
> I tried what was mentioned above:
> sudo -E rpi-eeprom-config --edit
> USB_MSD_PWR_OFF_TIME=0
> 
> also in my config.txt I already had and have:
> program_USB_boot_mode=1
> 
> in cmdline.txt I have also tried changing the root=PARTUUID=85f65e63-02 to root=/dev/sda2.
> 
> All to no avail.
> 
> To be honest I am not sure how to reflash the eeprom back to 09-30 version, if that would help.
> 
> The Orico drive now also has issues being recognized on my Mac; diskutil list shows no drive....(yesterday the mac first didn't find it, and then it does; the raspberry pi does see it clearly)
> Using an SD I have no issues getting into the Rasperry Pi. Is it possible that the EEPROM update somehow messed up the Orico firmware? lsusb and lsblk do show that the Orico SSD is connected though.

The 3B+ doesn’t have a bootloader in SPI flash. This repo is specific to Pi4

**Comment by THX1180** [comment]
> Hello, I have a raspberry 3B+ with the Orico BH100, which as the:
> Bus 001 Device 005: ID 152d:0567 JMicron Technology Corp. / JMicron USA Technology Corp. JMS567 SATA 6Gb/s bridge
> 
> Until yesterday it worked fine, however after an update it won't boot anymore.
> I tried what was mentioned above:
> sudo -E rpi-eeprom-config --edit
> USB_MSD_PWR_OFF_TIME=0
> 
> also in my config.txt I already had and have:
> program_USB_boot_mode=1
> 
> in cmdline.txt I have also tried changing the root=PARTUUID=85f65e63-02 to root=/dev/sda2.
> 
> All to no avail.
> 
> To be honest I am not sure how to reflash the eeprom back to 09-30 version, if that would help.
> 
> The Orico drive now also has issues being recognized on my Mac; diskutil list shows no drive....(yesterday the mac first didn't find it, and then it does; the raspberry pi does see it clearly)
> Using an SD I have no issues getting into the Rasperry Pi. Is it possible that the EEPROM update somehow messed up the Orico firmware? lsusb and lsblk do show that the Orico SSD is connected though.

Check out my answer... I have the Pi 4 but could be useful for you.

**Comment by timg236** [comment]
Try the latest stable release (raspi-config advanced) 2021-01-16 if that doesn’t work raise a new bug.
Closing this because the original problem fixed. From now no one bug report per hardware configuration

**Comment by bschatzow** [comment]
> > Not sure if this is a good place to post or not. My issue is I use Home Assistant on the rpi-4 4 GB. My system boots off the SSD drive an has no issues with OS 5.2 any updates after this causes my system to freeze in 2 hours to 2 days. No useful information in the logs. CPU use rises, Memory, and everything else is normal. My Controller is a StarTech (USB3S2SAT3CB) and the SSD is a Kingston 120GB A400 SATA 3 2.5" Internal SSD SA400S37. Many comments in [home-assistant/operating-system#1119](https://github.com/home-assistant/operating-system/issues/1119). I am willing to help test. Let me know what addition information is needed and what I should try.
> > Thanks.
> 
> That sounds like a Linux kernel update issue because the Linux kernel doesn't depend upon the bootloader USB stack.

Any idea where I can ask for help?  Thanks.

**Comment by sstefanowski** [comment]
Maybe it could be interesting for other users but I have also ORICO BH100 and encountered major issues trying to boot a Raspberry PI 3B+ with it. I think this drive may be poorly designed. 

1. Is it possible that ORICO BH100 have some power issues? My RPi3B+ was booting from this drive only when connected to the powerfull source - it was not booting from standard RPi Power source giving 2.5A but it was booting successfully when RPi was powered from Huawei charger giving as much as 4.5A(!). It's quite strange anyway as I thought RPi USB power is limited by the board. But this observation was absolutely repeatable issue.
2. Second - I encountered random read failures on this SSD. It was like after some time the random incorrect bytes were read from files on SSD making some services (Home Assistant mostly unusable). The strange thing was after reboot the file previously corrupted was read correctly and after some time the read error appeared in another place. I tried to change USB cable, power source and RPi and it was still the issue.
3. Strange is when Orico BH100 is powered by my Win10 laptop there is no issue, but when I connected this drive to my Networked Media Player (Dune HD - based on Linux) then the movies from this disk were rendered on the TV screen with visual artefacts - I assume it just confirms the byte-read-failure issues on this disk.
So maybe this is a faulty item? Or problems are related to USB on Linux? (as I mentined DuneHD probably has the same issues)? Or power problem with BH100 case? No idea.... Any hints?

Final of the story... I replaced Orico BH100 with Kingston A400 120GB + IBox HD-05 (with JMicron 578 inside!) and it works fine. RPi 3B+ boots successfully even on Standard RPi power giving 2.5A. I even managed to enable FSTRIM on this disk but... hmmmm... it required downgrading firmware of the HD-05 case (JMicron 587) to 5.00.08 in my case (the process is to be found on the web)...

Now... I think I will try to replace JMicron in BH100 with ASM1153E as advised by @THX1180.

**Comment by stek29** [comment]
@sstefanowski what version of VL805 firmware were you using on your latest test?

**Comment by pepsonEL** [comment]
Hi
Anybody can help me please. I have Orico case for mSATA SSD on USB3.0 and i try to boot from it my RPI 4 8GB. no found solution , also try many many FW for JMS578 and nothing. Also try this solution:
Solution: with the command "sudo -E rpi-eeprom-config --edit" you edit the EEPROM BOOTLOADER then you the next configuration: USB_MSD_PWR_OFF_TIME=0 but also not help me.... My RPI no boot. I have in my RPI eeprom latest FW from 17.03.2021.... BETA. Please help me

**Comment by bougs93** [comment]
I have 3 sets of ORICO M2PF-C models. Raspberry Pi 4 also has 4 sets.
It uses jms580 chipset.
It cannot be used because it does not boot.
Please help me.

**Comment by bougs93** [comment]
It's my experience.

ORICO Model: M2PF-C3 using JMS580 chipset works fine.
You need to install the RaspbianOS image from scratch.
If you copy the already installed SD image to the SSD, it will not be recognized.

Here's what I found out over three days.

**Comment by var1ableX** [comment]
SOLVED: I’d been trying to get SSD to boot all day. Believe SABRENT enclosure is the problem.
What I tried:
1. Created SSD image on MacOS using RPi Imager (tried lite, and full 32 bit images). Didn’t boot.
2. Tried to copy MicroSD image onto SSD using Copy SD function of RPi GUI. Got Aborting Drives Changed error.

Finally I saw articles suggesting SABRENT was the problem so I switched enclosure to a UGREEN enclosure I was using for another drive. Everything worked.

I did come to realize that an intermittent green led that constantly flashes is not a sign that anything is wrong. I’m guessing it just means Pi is waiting for data. This would be different if it’s flashing 3x 4x 5x etc. But if it’s just flashing constantly then it’s ok.

**Comment by tiagoboldt** [comment]
This can be solved using by appending the following argument to `/boot/cmdline.txt`.:

> usb-storage.quirks=152d:0578:u

Confirm with `lsusb` if your enclosure has that USB product ID, you might need to adjust the argument to match your enclosure version.

**Comment by lurch** [comment]
@tiagoboldt Yeah that's mentioned in the documentation [here](https://www.raspberrypi.com/documentation/computers/raspberry-pi.html#hardware-compatibility).

**Comment by eebssk1** [comment]
I know this is quietly old but I still want to notice that JMicron chips do not work very well in many use cases. I just fell into the pit today.You may want to get a enclosure with rtl or asm chips instead.

**Comment by tablatronix** [comment]
Yeah I gave up and wound up using nvme in a sabrent usb c case

**Comment by Zyzonix** [comment]
That's actually true, but those issues are only present when using USB 3. Think Raspberry Pi doesn't provide enough power (also with the official PS) to power the external NVMe and the PCIe bride. If you aren't in need for high transferrates just use USB 2, it's a bit slower, but works fine. @tablatronix @eebssk1

**Comment by eebssk1** [comment]
> That's actually true, but those issues are only present when using USB 3. Think Raspberry Pi doesn't provide enough power (also with the official PS) to power the external NVMe and the PCIe bride. If you aren't in need for high transferrates just use USB 2, it's a bit slower, but works fine. @tablatronix @eebssk1 

Might be incompatible instructions. Usb3/2 their difference are only max suppliable current. The disks use constant current when R/W.
The only difference is USB2 only support BOT  as transfer method while USB3 support both BOT and UASP(preferred.)
BOT is more compatible as it's the first file transfer protocol born with USB.

[Other information(off topic): It seems that JMS58* is incompatible with USB3.1 HUB that uses genesys logic chip.
I returned the JMS580b enclosure and then replaced it with a enclosure with RTL9201R(even cheapier), everything works fine now.]

**Comment by tablatronix** [comment]
```
Bus 002 Device 003: ID 0bda:9210 Realtek Semiconductor Corp. RTL9210 M.2 NVME Adapter

 sudo hdparm -Tt /dev/sda2

/dev/sda2:
 Timing cached reads:   2106 MB in  1.99 seconds = 1055.70 MB/sec
 Timing buffered disk reads: 1034 MB in  3.01 seconds = 344.05 MB/sec
 ...
 Timing O_DIRECT disk reads: 996 MB in  3.00 seconds = 331.75 MB/sec
```

**Comment by terrafirma2021** [comment]
> This can be solved using by appending the following argument to `/boot/cmdline.txt`.:
> 
> > usb-storage.quirks=152d:0578:u
> 
> Confirm with `lsusb` if your enclosure has that USB product ID, you might need to adjust the argument to match your enclosure version.

Thanks, this is the solution for this controller,
